bluesky/plans.py

```diff
@@ -964,10 +964,18 @@ def relative_inner_product_scan(detectors, num, *args, per_step=None, md=None):
     :func:`bluesky.plans.inner_product_scan`
     :func:`bluesky.plans.scan_nd`
     """
-    _md = {'plan_name': 'relative_inner_product_scan'}
+    _md = {'plan_name': 'relative_inner_product_scan',
+           }
     _md.update(md or {})
     motors = [motor for motor, start, stop in partition(3, args)]
 
+    # update the hints separately for the best effort callback
+    # TODO : consider non-primary?
+    default_hints = {'dimensions': [((motors[0].name,), 'primary')]}
```

Comment:
The 'primary' here is what data stream these things are in.  It is probably better to ask the motor for it's hints.  The value we care about here is probably `motors[0].user_readback.name` which is most cases is the same as `motors[0].name` but does not have to be.  

We also want to list _all_ of the motors in the dimensions, not just the first one.  This should probably be something like

```python
[(motor.hints.get('fields', ()), 'pirmary') for motor in motors]

```

but I am not sure that that nesting is correct.

Reply:
sounds good. it should not be a list of motor names but a choice of one for a 1D plot see the modification. not sure how to make it look nicer in terms of list comprehension. @danielballan should have some say as to the general API, if this is the right place to put it